file/file.go

```diff
@@ -0,0 +1,60 @@
+package file
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+)
+
+// GroupVarsFile ...
+type GroupVarsFile struct {
+	Path string
+}
+
+// NewGroupVarsFile ... factory/ctor
+func NewGroupVarsFile(path string) GroupVarsFile {
+	f := GroupVarsFile{
+		Path: path,
+	}
+	return f
+}
+
+// HandleGroup ...
+func (gvf GroupVarsFile) HandleGroup(group string) (map[string]interface{}, error) {
+	groupFile := fmt.Sprintf("%s.yml", group)
+	fileName := filepath.Join(gvf.Path, groupFile)
+
+	info, err := os.Stat(fileName)
+	if os.IsNotExist(err) {
+		return nil, err
+	}
+
+	if info.IsDir() {
+		return nil, fmt.Errorf("Path is a directory: %s", fileName)
+	}
+
+	buf, err := readFile(fileName)
+	if err != nil {
+		return nil, err
+	}
+
+	groupFileYaml := make(map[string]interface{})
+	err = yaml.Unmarshal(buf, &groupFileYaml)
+	if err != nil {
+		return nil, err
+	}
+
+	return groupFileYaml, nil
+}
+
+// ReadFile ... from disk
+func readFile(path string) ([]byte, error) {
+	buf, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	return buf, nil
```

Comment:
isn't this exactly the same as `return ioutil.ReadFile(path)`? I.e. I think we can just use that directly